vortex-array/src/array/bool/compute/boolean.rs

```diff
@@ -0,0 +1,36 @@
+use arrow_arith::boolean;
+use arrow_array::cast::AsArray;
+use vortex_error::VortexResult;
+
+use crate::array::bool::BoolArray;
+use crate::arrow::FromArrowArray;
+use crate::compute::{AndFn, OrFn};
+use crate::{Array, ArrayData, IntoArray, IntoCanonical};
+
+impl OrFn for BoolArray {
+    fn or(&self, array: &Array) -> VortexResult<Array> {
+        let lhs = self.clone().into_canonical()?.into_arrow();
+        let lhs = lhs.as_boolean();
+
+        let rhs = array.clone().into_canonical()?.into_arrow();
+        let rhs = rhs.as_boolean();
```

Comment:
Should this maybe dispatch on the right hand side instead of converting to arrow?

Reply:
definitely possible, but I think we might be missing some larger abstraction here to solve the whole left/right issue